src/storage/index/IndexPolicyMaker.cpp

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/index/IndexPolicyMaker.h"
+#include "common/base/NebulaKeyUtils.h"
+
+namespace nebula {
+namespace storage {
+cpp2::ErrorCode IndexPolicyMaker::preparePolicy(const std::string &filter) {
+    auto ret = decodeExpression(filter);
+    if (ret != cpp2::ErrorCode::SUCCEEDED) {
+        return ret;
+    }
+    /**
+     * Traverse the expression tree and
+     * collect the relationship for execution policies.
+     */
+    ret = traversalExpression(exp_.get());
+    return ret;
+}
+
+cpp2::ErrorCode IndexPolicyMaker::decodeExpression(const std::string &filter) {
+    cpp2::ErrorCode code = cpp2::ErrorCode::SUCCEEDED;
+    auto expRet = Expression::decode(filter);
+    if (!expRet.ok()) {
+        VLOG(1) << "Can't decode the filter " << filter;
+        return cpp2::ErrorCode::E_INVALID_FILTER;
+    }
+    exp_ = std::move(expRet).value();
+    if (expCtx_ == nullptr) {
+        expCtx_ = std::make_unique<ExpressionContext>();
+    }
+    exp_->setContext(this->expCtx_.get());
+    auto status = exp_->prepare();
+    if (!status.ok()) {
+        return cpp2::ErrorCode::E_INVALID_FILTER;
+    }
+    return code;
+}
+
+void IndexPolicyMaker::buildPolicy() {
+    prefix_.reserve(256);
+    decltype(operatorList_.size()) hintNum = 0;
+    bool hasStr = false;
+    for (auto& col : index_->get_fields()) {
+        auto it = std::find_if(operatorList_.begin(), operatorList_.end(),
+                               [&col] (const auto& tup) {
+                                   return col.get_name() == std::get<0>(tup);;
+                               });
+        if (it != operatorList_.end()) {
+            if (std::get<2>(*it) == RelationalExpression::Operator::EQ) {
```

Comment:
We don't handle ">, <, >=, <=" ?

Reply:
> We don't handle ">, <, >=, <=" ?

Yes , currently use expression to handle such operations  ">, <, >=, <=" .
 In fact, the best way is to handle with it in a subset of range.

Reply:
Yes,  full scan the whole index is too costly.